src/backend/distributed/commands/role.c

```diff
@@ -1181,11 +1198,16 @@ PreprocessGrantRoleStmt(Node *node, const char *queryString,
 List *
 PostprocessGrantRoleStmt(Node *node, const char *queryString)
 {
-	if (!EnableCreateRolePropagation || !IsCoordinator() || !ShouldPropagate())
+	if (!EnableCreateRolePropagation || !ShouldPropagate())
 	{
 		return NIL;
 	}
 
+	if (!IsCoordinator())
+	{
+		EnsureCoordinatorIsInMetadata();
+	}
```

Comment:
This check seems common enough that we should probably make it a single function call. Maybe call it something like `EnsurePropagationToCoordinator()`